app/src/main/java/com/nutomic/syncthingandroid/activities/FirstStartActivity.java

```diff
@@ -94,11 +123,28 @@ public boolean onTouch(View v, MotionEvent event) {
             }
         });
 
-        // Layouts of all welcome slides
-        mLayouts = new int[]{
-                R.layout.activity_firststart_slide1,
-                R.layout.activity_firststart_slide2,
-                R.layout.activity_firststart_slide3};
+        // Add welcome slides to be shown.
+        int[] colorsActive = getResources().getIntArray(R.array.array_dot_active);
+        int[] colorsInactive = getResources().getIntArray(R.array.array_dot_inactive);
+        int slideIndex = 0;
+        mSlides = new Slide[
+                1 +
+                (showSlideStoragePermission ? 1 : 0) +
+                (showSlideLocationPermission ? 1 : 0) +
+                (showSlideKeyGeneration ? 1 : 0)
+        ];
```

Comment:
Just instantiate this to 4 or 10 or whatever the amount you need is, and just check for nulls.

Reply:
Or use an linked list or something.